apps/prairielearn/assets/scripts/lib/confirmOnUnload.ts

```diff
@@ -0,0 +1,37 @@
+function getQuestionFormData(form: HTMLFormElement): string {
+  // Cast FormData since TS does not support this parameter,
+  // see https://github.com/microsoft/TypeScript/issues/30584
+  const formData = new URLSearchParams(new FormData(form) as any);
+  formData.delete('__csrf_token');
+  form.querySelectorAll('[data-skip-unload-check]').forEach((input: HTMLInputElement) => {
+    if (input.name) formData.delete(input.name);
+  });
+  return formData.toString();
+}
+
+export function saveQuestionFormData(form: HTMLFormElement) {
+  if (!form) return;
```

Comment:
`form` cannot be falsy per the type of `form` in the function signature..

Reply:
This can be called with null, when in assessment overview. Updated function signature in 36a4224f52a3a60316985982cc9a37df86a8b4a9.